pkg/registry/thirdpartyresource/etcd/etcd_test.go

```diff
@@ -80,26 +80,24 @@ func TestCreate(t *testing.T) {
 func TestUpdate(t *testing.T) {
 	storage, fakeClient := newStorage(t)
 	test := resttest.New(t, storage, fakeClient.SetError)
-	key, err := storage.KeyFunc(test.TestContext(), "foo")
-	if err != nil {
-		t.Fatal(err)
-	}
-	key = etcdtest.AddPrefix(key)
-	fakeClient.ExpectNotFoundGet(key)
-	fakeClient.ChangeIndex = 2
-	rsrc := validNewThirdPartyResource("foo")
-	existing := validNewThirdPartyResource("exists")
-	existing.Namespace = test.TestNamespace()
-	obj, err := storage.Create(test.TestContext(), existing)
-	if err != nil {
-		t.Fatalf("unable to create object: %v", err)
-	}
-	older := obj.(*expapi.ThirdPartyResource)
-	older.ResourceVersion = "1"
 	test.TestUpdate(
-		rsrc,
-		existing,
-		older,
+		// valid
+		validNewThirdPartyResource("foo"),
+		func(ctx api.Context, obj runtime.Object) error {
```

Comment:
I discussed this with @lavalamp.
We cannot use Get() directly here since there could be write-only resources (which support Create(), but not Get()).
But we should not require etcd_tests to pass these get, set functions. The resttest framework itself can have these functions.
That will eliminate the boilerplate code of passing those functions in all our pkg/registry/*/etcd/etcd_tests.go files.
Will try to come up with a PR with that change and we can see how it looks.